### PR TITLE
fix #4699 remove cache from history when offline log is deleted

### LINKED ISSUE #4699: Cache not removed from history after offline log deleted
2015.02.21-NB

- Log a cache offline
- Verify it is added to the history
- Open the cache again and delete the offline log by Menu - Log offline - Delete (or whatever the menu is called in english)

The cache keeps showing up in the history permanently (also after app restart)

### PR DESCRIPTION
**Pull request by pstorch**
As #4086 might take longer, I'll send a fix for this separately.

When removing an offline log it also sets the `visiteddate` to 0 to remove it from History.

### CODE REVIEW

File: main/src/cgeo/geocaching/storage/DataStore.java [hunk 1]
```diff
@@ -2652,7 +2652,9 @@ public static void clearLogOffline(final String geocode) {
 
         init();
 
-        database.delete(dbTableLogsOffline, "geocode = ?", new String[]{geocode});
+        final String[] geocodeWhereArgs = {geocode};
+        database.delete(dbTableLogsOffline, "geocode = ?", geocodeWhereArgs);
+        database.execSQL(String.format("UPDATE %s SET visiteddate = 0 where geocode = ?", dbTableCaches), geocodeWhereArgs);
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/storage/DataStore.java]: `WHERE` should be in uppercase (several instances) as it is a SQL keyword.

Comment (pstorch) [inline main/src/cgeo/geocaching/storage/DataStore.java]: Fixed these. 
There are some more places in this class where SQL keywords are lowercase. Maybe I go through them in a separate PR in one go.

Comment (samueltardieu) [inline main/src/cgeo/geocaching/storage/DataStore.java]: That would be great indeed.

**Comment by samueltardieu** [comment]
retest this please
